src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -345,6 +346,58 @@ SIMPLE_MATCHER_PASS_DEFINITION(EliminateGather,
                                ov::op::v7::Gather,
                                ov::op::v8::Gather);
 
+pass::EliminateReduceReshape::EliminateReduceReshape() {
+    MATCHER_SCOPE(EliminateReduceReshape);
+    using namespace pass::pattern;
+    auto axes = wrap_type<ov::op::v0::Constant>();
+    auto reduce_pattern = wrap_type<ov::op::util::ArithmeticReductionKeepDims>({any_input(), axes});
+    auto requested_shape_pattern = wrap_type<ov::op::v0::Constant>();
+    auto reshape_pattern =
+        wrap_type<ov::op::v1::Reshape>({reduce_pattern, requested_shape_pattern}, consumers_count(1));
+
+    matcher_pass_callback callback = [=](Matcher& m) {
+        auto pattern_map = m.get_pattern_map();
+        auto reshape_node = m.get_match_root();
+        auto reduce_node = pattern_map.at(reduce_pattern);
+
+        auto reshape = ov::as_type_ptr<ov::op::v1::Reshape>(reshape_node);
+        auto reduce = ov::as_type_ptr<ov::op::util::ArithmeticReductionKeepDims>(reduce_node);
+        if (!reshape || !reduce) {
+            return false;
+        }
+
+        auto in_rank = reshape->get_input_partial_shape(0).rank();
+        auto out_rank = reshape->get_output_partial_shape(0).rank();
+        if (in_rank.is_dynamic() || out_rank.is_dynamic() || in_rank.get_length() != out_rank.get_length()) {
```

Comment:
Should we check is_dynamic() here or move this check (only is it static) to the pattern?

Reply:
I would prefer to have this check here before the actual usage " in_rank.get_length() != out_rank.get_length()"

Reply:
```suggestion
        if (in_rank.is_dynamic() || (in_rank.get_interval() != out_rank.get_interval())) {
```
Maybe this will work also? 
`get_length()` do `is_dynamic` check again.

Reply:
It might work, but it's not that obvious from the code
I believe "in_rank.is_dynamic() || out_rank.is_dynamic() || in_rank.get_length() != out_rank.get_length()" is more human-readable